### PR TITLE
Updated JsonRpc Api to v0.3.0-rc1

### PR DESCRIPTION
**Pull request by bartekryba**
Updated JsonRpc api to support [0.3.0-rc1](https://github.com/starkware-libs/starknet-specs/releases/tag/v0.3.0-rc1) spec.

## Usage related changes

- Added Cairo1 class support
- Added support for declare v2
- It's now possible to estimate fee for multiple transactions
- Felt validation got updated (No more 0x0 prefixed felts)

## Development related changes

- For `run_devnet_in_background` method, the flag `--cairo-compiler-manifest` will be set by default if it wasn't explicitly set.

## Checklist:

- [x] Applied formatting - `./scripts/format.sh`
- [x] No linter errors - `./scripts/lint.sh`
- [x] Performed code self-review
- [x] Rebased to the last commit of the target branch (or merged it into my branch)
- [x] Documented the changes
- [ ] Linked the issues which this PR resolves
- [x] Updated the tests
- [x] All tests are passing - `./scripts/test.sh`

### CODE REVIEW

File: page/docs/guide/json-rpc-api.md [hunk 1]
```diff
@@ -1,10 +1,10 @@
----
+--
```
Comment (drknzz) [inline page/docs/guide/json-rpc-api.md]: Doesn't this break the file?

File: starknet_devnet/blueprints/rpc/utils.py [hunk 2]
```diff
@@ -75,15 +75,25 @@ async def assert_block_id_is_valid(block_id: BlockId) -> None:
...
     The value can be base 10 integer, base 10 string or base 16 string
     """
     if isinstance(value, str):
         value = int(value) if value.isnumeric() else int(value, 16)
 
     if value == 0:
-        return "0x00"
-    return "0x0" + hex(value).lstrip("0x")
+        return "0x0"
+    return "0x" + hex(value).lstrip("0x")
```
Comment (drknzz) [inline starknet_devnet/blueprints/rpc/utils.py]: Isn't this just `hex(value)` now?

File: starknet_devnet/blueprints/rpc/utils.py [hunk 3]
```diff
@@ -101,7 +111,7 @@ def rpc_root(root: str) -> Felt:
     """
     Convert 0 prefixed root to 0x prefixed root
     """
-    return "0x0" + (root.lstrip("0") or "0")
+    return "0x" + (root.lstrip("0") or "0")
```
Comment (drknzz) [inline starknet_devnet/blueprints/rpc/utils.py]: Could `rpc_felt` be used here? It seems that root is defined as felt in the spec

File: test/rpc/rpc_utils.py [hunk 4]
```diff
@@ -115,4 +115,4 @@ def is_felt(value: str) -> bool:
     """
     Check whether value is a Felt
     """
-    return bool(re.match(r"^0x0[a-fA-F0-9]{1,63}$", value))
+    return bool(re.match(r"^0x[a-fA-F0-9]{1,63}$", value))
```
Comment (drknzz) [inline test/rpc/rpc_utils.py]: The felt regex seems to have been changed
```
"pattern": "^0x(0|[a-fA-F1-9]{1}[a-fA-F0-9]{0,62})$"
```

File: test/rpc/test_data/get_events.py [hunk 5]
```diff
@@ -45,7 +45,8 @@ def create_get_events_filter(
     :return: `filter` param matching the specification.
     """
     if keys is None:
-        keys = [rpc_felt(FEE_CHARGED_EVENT_KEY)]
+        keys = []
```
Comment (drknzz) [inline test/rpc/test_data/get_events.py]: Was this omitted on purpose?

File: starknet_devnet/blueprints/rpc/misc.py [hunk 6]
```diff
@@ -41,11 +41,20 @@ def check_address(address, event):
...
 
 
-def check_keys(keys, event):
+def check_keys(keys: List[List[Felt]], event):
     """
     Check keys.
     """
-    return bool(set(event.keys) & set(keys))
+    if not keys:
+        return True
```
Comment (drknzz) [inline starknet_devnet/blueprints/rpc/misc.py]: Wouldn't the below code just return `True` if keys were empty?

Comment (drknzz) [reply starknet_devnet/blueprints/rpc/misc.py]: Or can keys be `None`? If so, let's maybe explicitly check `if keys is None` and also add `Optional` to the typing

File: starknet_devnet/blueprints/rpc/misc.py [hunk 7]
```diff
@@ -139,7 +148,11 @@ async def get_events(
         ) from ex
 
     address = filter.get("address")
-    keys = [int(k, 0) for k in filter.get("keys")]
+
+    if filter.get("keys"):
+        keys = [[int(key, 0) for key in keys] for keys in filter.get("keys")]
+    else:
+        keys = None
```
Comment (drknzz) [inline starknet_devnet/blueprints/rpc/misc.py]: Similar situation here, the `else` branch would execute in both cases when `keys` are not specified, and when they are an empty list.

File: starknet_devnet/blueprints/rpc/utils.py (hunk 2 above)
Comment (bartekryba) [reply starknet_devnet/blueprints/rpc/utils.py]: yeah 😅

File: test/rpc/test_data/get_events.py (hunk 5 above)
Comment (bartekryba) [reply test/rpc/test_data/get_events.py]: Updated it

**Pull request closed without merging**